### PR TITLE
New submission flow: Update test result pending screen (EXPOSUREAPP-3738)

### PR DESCRIPTION
**Pull request by AlexanderAlferov**
Updated test result pending steps and simple step entry custom view.

Test_result_pending_screenshot

### CODE REVIEW

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/view/SimpleStepEntry.kt [hunk 1]
```diff
@@ -25,6 +26,8 @@ class SimpleStepEntry @JvmOverloads constructor(
 
             simple_step_entry_body.text =
                 getText(R.styleable.SimpleStepEntry_simple_step_entry_text)
+
+            if (simple_step_entry_body.text.isEmpty()) simple_step_entry_body.visibility = View.GONE
```
Comment (d4rken) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/view/SimpleStepEntry.kt:30]: Can we change this to "findViewById" or ViewBinding in this PR? Kotlin synthetics will be removed in a future Android update.

https://developer.android.com/topic/libraries/view-binding/migration

File: Corona-Warn-App/src/main/res/layout/include_submission_test_result.xml [hunk 2]
```diff
@@ -57,6 +57,17 @@
                 app:layout_constraintStart_toStartOf="@+id/guideline_start"
                 app:layout_constraintTop_toBottomOf="@+id/submission_test_result_subtitle" />
 
+            <de.rki.coronawarnapp.ui.submission.consentstatus.ConsentStatusView
+                android:id="@+id/consent_status"
+                android:layout_width="@dimen/match_constraint"
+                android:layout_height="wrap_content"
+                android:layout_marginBottom="@dimen/spacing_medium"
+                android:focusable="true"
+                android:visibility="@{FormatterSubmissionHelper.formatTestResultPendingStepsVisible(uiState.deviceUiState)}"
```
Comment (d4rken) [inline Corona-Warn-App/src/main/res/layout/include_submission_test_result.xml:66]: We should refactor the FormatterSubmissionHelper at some point.
Not a blocker for this PR, just a note.

**Review by d4rken — CHANGES_REQUESTED** [review]
Looks good, but...
The consent view padding does not match the mockups, needs less top margin and more bottom margin.

The line connecting the steps also seems to be thinner than on the mockups, but that's not new in this PR :thinking:

**Comment by AlexanderAlferov** [comment]
> Looks good, but...
> The consent view padding does not match the mockups, needs less top margin and more bottom margin.
> 
> The line connecting the steps also seems to be thinner than on the mockups, but that's not new in this PR 🤔
Yes, mockups are not 100% compliant right now with our guideline, so there could be slight differences.

**Comment by ralfgehrer** [comment]
> > Looks good, but...
> > The consent view padding does not match the mockups, needs less top margin and more bottom margin.
> > The line connecting the steps also seems to be thinner than on the mockups, but that's not new in this PR 🤔
> 
> Yes, mockups are not 100% compliant right now with our guideline, so there could be slight differences.

In the screenshot, however, the padding on top of the consent is larger than the one below.

**Comment by AlexanderAlferov** [comment]
> > > Looks good, but...
> > > The consent view padding does not match the mockups, needs less top margin and more bottom margin.
> > > The line connecting the steps also seems to be thinner than on the mockups, but that's not new in this PR 🤔
> > 
> > 
> > Yes, mockups are not 100% compliant right now with our guideline, so there could be slight differences.
> 
> In the screenshot, however, the padding on top of the consent is larger than the one below.

True, added padding as was discussed.
Vuzix View_20201202_140332_125

**Review by ralfgehrer — APPROVED** [review]

File: Corona-Warn-App/src/main/res/layout/include_submission_test_result.xml [hunk 3]
```diff
@@ -53,6 +53,18 @@
                 app:layout_constraintStart_toStartOf="@+id/guideline_start"
                 app:layout_constraintTop_toBottomOf="@+id/submission_test_result_subtitle" />
 
+            <de.rki.coronawarnapp.ui.submission.consentstatus.ConsentStatusView
+                android:id="@+id/consent_status"
+                android:layout_width="@dimen/match_constraint"
+                android:layout_height="wrap_content"
+                android:layout_marginBottom="@dimen/spacing_medium"
+                android:paddingBottom="@dimen/spacing_small"
```
Comment (ralfgehrer) [inline Corona-Warn-App/src/main/res/layout/include_submission_test_result.xml:61]: Probably needs to be medium afterall.

**Review by Oliver-Zimmerman — APPROVED** [review]

File: Corona-Warn-App/src/main/res/layout/include_submission_test_result.xml (hunk 2 above)
Comment (AlexanderAlferov) [reply Corona-Warn-App/src/main/res/layout/include_submission_test_result.xml:63]: For sure we should place it into refactoring list.

**Review by d4rken — APPROVED** [review]

**Review by janetback — APPROVED** [review]
UA approved

**Pull request merged**